fairmq/MemoryResources.h

```diff
@@ -42,70 +43,82 @@ class FairMQMemoryResource : public pmr::memory_resource
     /// buffer), e.g. pointer returned by std::vector::data() return nullptr if
     /// returning
     /// a message does not make sense!
-    virtual FairMQMessagePtr getMessage(void *p) = 0;
-    virtual void *setMessage(FairMQMessagePtr) = 0;
-    virtual FairMQTransportFactory *getTransportFactory() noexcept = 0;
+    virtual MessagePtr getMessage(void* p) = 0;
+    virtual void* setMessage(MessagePtr) = 0;
+    virtual TransportFactory* getTransportFactory() noexcept = 0;
     virtual size_t getNumberOfMessages() const noexcept = 0;
 };
 
+auto createMessage(TransportFactory* factory, std::size_t bytes, Alignment alignment) -> MessagePtr;
+
 /// This is the allocator that interfaces to FairMQ memory management. All
 /// allocations are
 /// delegated to FairMQ so standard (e.g. STL) containers can construct their
 /// stuff in
 /// memory regions appropriate for the data channel configuration.
-class ChannelResource : public FairMQMemoryResource
+template<std::size_t MinAlignment>
+class AlignedChannelResource : public FairMQMemoryResource
```

Comment:
is the name change necessary? Since the default behavior does not change (or does it?), signatures do not change (as far as I can see) it seems to be the same animal, just a better behaved one :)

Reply:
If I don't change the name, it would not let me do the name alias: `using ChannelResource = ChannelResource<0>;` - do you know how to make it work?

Reply:
indeed, good question :)

Reply:
```c++
template <int alignment = 0>
struct ChannelResource {
};
```

no?